crates/iceberg/src/arrow/schema.rs

```diff
@@ -382,12 +382,15 @@ impl ArrowSchemaVisitor for ArrowSchemaConverter {
             DataType::Time64(unit) if unit == &TimeUnit::Microsecond => {
                 Ok(Type::Primitive(PrimitiveType::Time))
             }
-            DataType::Timestamp(unit, None) if unit == &TimeUnit::Microsecond => {
+            DataType::Timestamp(unit, None)
+                if unit == &TimeUnit::Microsecond || unit == &TimeUnit::Nanosecond =>
+            {
                 Ok(Type::Primitive(PrimitiveType::Timestamp))
```

Comment:
I don't think this is correct, in the case of the `Nanosecond` Arrow `DataType`, we want to map to `TimestampNs`, otherwise we'll be doing downcasting (that can be okay, as long as the user explicitly wants it, in PyIceberg we control this by explicitly setting a property).

Relevant types:

https://github.com/apache/iceberg-rust/blob/ce040585b647f8afaccb9f89d3adcbb8c9ec88f7/crates/iceberg/src/spec/datatypes.rs#L369-L370

Reply:
Ah! I was looking at the wrong thing when I was wiring this up, my fault :facepalm: 

I've adjusted these types to the `*Ns` variants :+1:

Reply:
No problem at all, thanks for adding this 🙌